jsx/FilterableDataTable.js

```diff
@@ -0,0 +1,37 @@
+import React, {Component} from 'react';
+
+import DynamicDataTable from 'jsx/DynamicDataTable';
+import FilterForm from 'jsx/FilterForm';
+
+class FilterableDataTable extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      filters: {}
+    };
+    this.updateFilter = this.updateFilter.bind(this);
+  }
+
+  updateFilter(filter) {
+    this.setState({filters: filter});
+  }
+
+  render() {
+    return (
+      <div>
+        <FilterForm
+          name={this.props.name + '_filter'}
+          id={this.props.name + '_filter_form'}
+          ref={this.props.name + 'Filter'}
```

Comment:
we are trying to remove all string refs as its getting deprecated so should probably change this to a callback method

Reply:
Is there an example somewhere of how to do this? I based this on (IIRC) the media module and just replaced the module name with a prop.

Reply:
https://github.com/aces/Loris/pull/4019 and https://reactjs.org/docs/refs-and-the-dom.html#callback-refs

Reply:
better yet, don't use refs at all. the use of refs in the other modules was used due to a lack of an appropriate common parent component between the filterform and the data table. with this component it looks like the use of refs for filter clearing can be deprecated altogether

Reply:
I agree with @davidblader but it implies that the datatable should take data as props. Fortunately, the StaticDataTable does exacly that